docs/monitoring-usage.rst

```diff
@@ -278,3 +278,112 @@ follows::
+which similarly requires that its available labels have values specified.
+
+>>> from gcloud import monitoring
+>>> import datetime
+>>> client = monitoring.Client()
+>>> label = LabelDescriptor('response_code', LabelValueType.INT64,
+    ...                         description='HTTP status code')
+>>> metric_descriptor = = client.metric_descriptor(
+                                 'custom.googleapis.com/my_app/response_count',
+                                 metric_kind=MetricKind.CUMULATIVE,
+                                 value_type=ValueType.INT64,
+                                 labels=[label],
+                                 description='Cumulative count of HTTP responses.')
+>>> if not metric_descriptor.exist():
+>>>     metric_descriptor.create()
+>>> # Create a Resource by its resource name, assign labels as dict
+>>> resource = client.resource('gce_instance',
+...                            labels={
+...                                'instance_id': 'my-instance',
+...                                'zone': 'us-central1-f'
+...                             })
+>>> # Create a Metric using its metric descriptor parametrized with labels
+>>> metric = client.metric(type=metric_descriptor.type, labels={
+...                                              'response_code': 404
+...                                               }
+
+It's also possible to assign labels to Metrics and Resources with keyword arguments::
+
+>>> metric = client.metric(type=metric_descriptor.type, response_code=404)
+>>> resource = client.resource('gce_instance', instance_id='my-instance', zone='us-central1-f')
+
+Please refer to the `Metrics`_ documentation for more information.
+
+With a Metric and Resource created, the :class:`~gcloud.monitoring.timeseries.TimeSeries`
+can be used to write  :class:`~gcloud.monitoring.timeseries.Point` values.
+
+When creating :class:`~gcloud.monitoring.timeseries.TimeSeries`, you specify points containing a
+value of the type that matches the *value_type* specified in the associated `MetricDescriptor`.
+The point must also contain a time interval. The point's time interval must be later than any
+points already created for the same :class:`~gcloud.monitoring.timeseries.TimeSeries` (where
+:class:`~gcloud.monitoring.timeseries.TimeSeries` with matching :class:`~gcloud.monitoring.metric
+.Metric` and :class:`~gcloud.monitoring.metric.Resource` types are considered the same). The end
+time of the interval must not be more than 24 hours in the past or more than five minutes in the
+future::
+
+    >>> start = datetime.utcnow()
+    >>> end = datetime.utcnow() - datetime.timedelta(minutes=5)
+    >>> value = get_num_404s_in_last_five_minutes() # get value from a hypothetical call
+    >>>
+    >>> client.write_timeseries(value, start_time=start, end_time=end,
+    ...                         metric=metric, resource=resource) # API call
```

Comment:
Ok, I was trying to get some opinions on it. The motivation for going in this direction was because it seems you have to consistently pass in Metric and Resource into every API call.  Seems if the common use case was to write points for a metric over time, and if those keep taking the same arguments every call, it would be better to have something to encapsulate those arguments. 

I agree it's a bit inconsistent, but it reflects the TimeSeries being slightly inconsistent at an API level- it's one of the few API calls where you call `create` multiple times to update it (most API objects are created once with a unique ID and after that only updated). TimeSeries should really have a write_point method of some sort, but it doesn't, so to do multiple writes to the same TimeSeries we can either copy the TimeSeries to a new object but replace the points (similar to what the `header` method does now but will need to add keyword argument magic), or keep recreating similar objects with the same arguments. Both seemed clumsy to me so I was putting an alternative out there but sounds like you're not a fan. I will mull it over and curious if Ken has an opinion, ultimately not a big deal either way and fine to get rid of it if you feel strongly.